azure-auth-helper/src/main/java/com/microsoft/azure/auth/LocalAuthServer.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.auth;
+
+import com.nimbusds.jose.util.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URI;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+
+public class LocalAuthServer {
+    private static transient boolean inited = false;
+    private static Object initLock = new Object();
+    private static String loginSuccessBodyTemplate;
+    private static String loginErrorBodyTemplate;
+    private final Semaphore semaphore = new Semaphore(0);
+    private final Server jettyServer;
+
+    private String code;
+    private String error;
+    private String errorDescription;
+
+    public LocalAuthServer() {
+        jettyServer = new Server();
+        final ServerConnector connector = new ServerConnector(jettyServer);
+        connector.setHost("localhost");
+        jettyServer.setConnectors(new Connector[]{ connector });
+        jettyServer.setHandler(new AbstractHandler() {
+            @Override
+            public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+                    throws IOException, ServletException {
+                initHtmlTemplate();
+                try {
```

Comment:
So we don't care about the path anymore???

Reply:
yes, the jetty server setup above is not using routes.